File: runtime/jvm/src/lox/LoxInstance.java
package lox;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * An instance's fields are open-ended (spring into existence on first
 * assignment — spec/03-types.md), so a plain map is the field table; no
 * per-class field layout exists. Identity equality (Lox default for this
 * type) comes for free from Object, so equals/hashCode are left alone.
 */
public final class LoxInstance {
    public final LoxClass klass;
    public final Map<String, Object> fields = new LinkedHashMap<>();

    public LoxInstance(LoxClass klass) {
        this.klass = klass;
    }
}

File: runtime/clr/src/LoxInstance.cs
using System.Collections.Generic;

namespace Lox;

/// <summary>
/// An instance's fields are open-ended (spring into existence on first
/// assignment - spec/03-types.md), so a plain map is the field table; no
/// per-class field layout exists. Identity equality (Lox default for this
/// type) comes for free from <see cref="object"/>, so Equals/GetHashCode
/// are left alone.
/// </summary>
public sealed class LoxInstance {
    public readonly LoxClass Klass;
    public readonly Dictionary<string, object> Fields = new();

    public LoxInstance(LoxClass klass) {
        Klass = klass;
    }
}
